gdrive

```diff
@@ -56,25 +56,21 @@ def init():
     service = build('drive', 'v3', credentials=creds)
     return service
 
-def get_filename(service, file_id):
+def get_metadata(service, file_id):
     """Gets the file name for a file ID"""
-    filename = 'gdrive-file'
     try:
-        file_metadata = service.files().get(fileId=file_id, fields='name').execute()
-        filename = file_metadata['name']
-    except Exception as err:
-        print('An error happened while getting file name.')
-        print(err)
-
-    print('Filename is "%s" for ID: %s.' % (filename, file_id))
-    return filename
+        fields = 'id, name, size, modifiedTime, modifiedByMeTime, owners'
+        return service.files().get(fileId=file_id, fields=fields).execute()
+    except:
```

Comment:
Check which exceptions this can raise and try to explicitly set them

Reply:
Catching googleapiclient.errors.HttpError now
In this case, I believe it's the only exception that can be thrown
Need to treat other errors on list, download, and upload